ansible_base/resource_registry/views.py

```diff
@@ -0,0 +1,74 @@
+from django.http import HttpResponseNotFound
+from django.shortcuts import get_object_or_404, redirect
+from rest_framework import permissions
+from rest_framework.decorators import action
+from rest_framework.response import Response
+from rest_framework.views import APIView
+from rest_framework.viewsets import GenericViewSet, mixins
+
+from ansible_base.resource_registry.models import Resource, ResourceType, service_id
+from ansible_base.resource_registry.registry import get_registry
+from ansible_base.resource_registry.serializers import ResourceListSerializer, ResourceSerializer, ResourceTypeSerializer, get_resource_detail_view
+
+
+class ResourceViewSet(
+    mixins.CreateModelMixin,
+    mixins.RetrieveModelMixin,
+    mixins.UpdateModelMixin,
+    mixins.DestroyModelMixin,
+    mixins.ListModelMixin,
+    GenericViewSet,
+):
+    """
+    Index of all the resources in the system.
+    """
+
+    queryset = Resource.objects.select_related("content_type__resource_type").all()
+    serializer_class = ResourceSerializer
+    permission_classes = [permissions.IsAdminUser]
```

Comment:
What I believe we need here is a permission class that corresponds exactly to something set by `JWTAuthentication`. I don't even expect that superusers can make changes here. It has to be made by a request that carries valid JWT authentication. It would make sense for superusers and auditors to be able to GET.

https://github.com/ansible/django-ansible-base/blob/devel/ansible_base/jwt_consumer/common/auth.py

@john-westcott-iv does this align with your thinking?